hardware_interface:
  joints:
    - bl_caster_l_wheel_joint
    - bl_caster_r_wheel_joint
    - bl_caster_rotation_joint
    - br_caster_l_wheel_joint
    - br_caster_r_wheel_joint
    - br_caster_rotation_joint
    - fl_caster_l_wheel_joint  
    - fl_caster_r_wheel_joint
    - fl_caster_rotation_joint
    - fr_caster_l_wheel_joint
    - fr_caster_r_wheel_joint
    - fr_caster_rotation_joint
    - head_pan_joint
    - head_tilt_joint
    - l_elbow_flex_joint
    - l_forearm_roll_joint
    - l_gripper_joint
    - l_gripper_l_finger_joint
    - l_gripper_l_finger_tip_joint
    - l_gripper_motor_screw_joint
    - l_gripper_motor_slider_joint
    - l_gripper_r_finger_joint
    - l_gripper_r_finger_tip_joint
    - l_shoulder_lift_joint
    - l_shoulder_pan_joint
    - l_upper_arm_roll_joint
    - l_wrist_flex_joint
    - l_wrist_roll_joint
    - laser_tilt_mount_joint
    - r_elbow_flex_joint
    - r_forearm_roll_joint
    - r_gripper_joint
    - r_gripper_l_finger_joint
    - r_gripper_l_finger_tip_joint
    - r_gripper_motor_screw_joint
    - r_gripper_motor_slider_joint
    - r_gripper_r_finger_joint
    - r_gripper_r_finger_tip_joint
    - r_shoulder_lift_joint
    - r_shoulder_pan_joint
    - r_upper_arm_roll_joint
    - r_wrist_flex_joint
    - r_wrist_roll_joint
    - torso_lift_joint
    - torso_lift_motor_screw_joint
generic_hw_control_loop/loop_hz: 100
generic_hw_control_loop/cycle_time_error_threshold: 0.01
r_arm_controller:
  type: position_controllers/JointTrajectoryController
  joints:
    - r_shoulder_pan_joint
    - r_shoulder_lift_joint
    - r_upper_arm_roll_joint
    - r_elbow_flex_joint
    - r_forearm_roll_joint
    - r_wrist_flex_joint
    - r_wrist_roll_joint
  gains:
    r_shoulder_pan_joint:
      p: 2400.0
      d: 18.0
      i: 800.0
      i_clamp: 4.0
      i_clamp_max: 4.0
      i_clamp_min: -4.0
    r_shoulder_lift_joint:
      p: 1200.0
      d: 10.0
      i: 700.0
      i_clamp: 4.0
      i_clamp_max: 4.0
      i_clamp_min: -4.0
    r_upper_arm_roll_joint:
      p: 1000.0
      d: 6.0
      i: 600.0
      i_clamp: 4.0
      i_clamp_max: 4.0
      i_clamp_min: -4.0
    r_elbow_flex_joint:
      p: 700.0
      d: 4.0
      i: 450.0
      i_clamp: 4.0
      i_clamp_max: 4.0
      i_clamp_min: -4.0
    r_forearm_roll_joint:
      p: 300.0
      d: 6.0
      i: 300.0
      i_clamp: 2.0
      i_clamp_max: 2.0
      i_clamp_min: -2.0
    r_wrist_flex_joint:
      p: 300.0
      d: 4.0
      i: 300.0
      i_clamp: 2.0
      i_clamp_max: 2.0
      i_clamp_min: -2.0
    r_wrist_roll_joint:
      p: 300.0
      d: 4.0
      i: 300.0
      i_clamp: 2.0
      i_clamp_max: 2.0
      i_clamp_min: -2.0
